src/plugins/console/public/application/hooks/use_data_init/use_data_init.test.ts

```diff
@@ -0,0 +1,201 @@
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should calls dispatch with new objects if no texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockedObject,
+    });
+  });
+
+  it('should calls dispatch with new objects if texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    const mockObj: MockObjectData = {
+      createdAt: 1643277939899,
+      updatedAt: 1643277939900,
+      text: 'Mocked data',
+    };
+
+    (useServicesContext as jest.Mock).mockReturnValue(callMockFuncWithArg([mockObj]));
+
+    renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockObj,
+    });
+  });
+
+  it('should update the states if the retry function is called', () => {
+    const setErrorMock = jest.fn();
+    const useErrorMock: UseStateMock = (state: any) => [state, setErrorMock];
+
+    const setDoneMock = jest.fn();
+    const useDoneMock: UseStateMock = (state: any) => [state, setDoneMock];
+
+    const setRetryTokenMock = jest.fn();
+    const useRetryTokenMock: UseStateMock = (state: any) => [state, setRetryTokenMock];
+
+    jest
+      .spyOn(React, 'useState')
+      .mockImplementationOnce(useErrorMock)
+      .mockImplementationOnce(useDoneMock)
+      .mockImplementationOnce(useRetryTokenMock);
+
+    const { result } = renderHook(() => useDataInit(), {});
+
+    result.current.retry();
+
+    expect(setErrorMock).toHaveBeenCalledWith(null);
+    expect(setDoneMock).toHaveBeenCalledWith(false);
+    expect(setRetryTokenMock).toHaveBeenCalledWith({});
+  });
+
+  it('should update the error state if case of getting an exception while loading data', async () => {
+    const error = new Error('Message');
+
+    (useServicesContext as jest.Mock).mockReturnValue(
+      callMockFuncWithArg(() => {
+        throw error;
+      })
+    );
+
+    const setDoneMock = jest.fn();
+    const setErrorMock = jest.fn();
+    const setRetryTokenMock = jest.fn();
+    const useDoneMock: UseStateMock = (state: any) => [state, setDoneMock];
+    const useErrorMock: UseStateMock = (state: any) => [state, setErrorMock];
+    const useRetryTokenMock: UseStateMock = (state: any) => [state, setRetryTokenMock];
+
+    jest.spyOn(React, 'useState').mockImplementationOnce(useErrorMock);
+    jest.spyOn(React, 'useState').mockImplementationOnce(useDoneMock);
+    jest.spyOn(React, 'useState').mockImplementationOnce(useRetryTokenMock);
+
+    renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(setErrorMock).toHaveBeenCalledWith(error);
+  });
+
+  it('should update the done state when we get into the finally block', async () => {
```

Comment:
This test is not covering the case, when `text.create`/`text.findAll` or `migrateToTextObjects` have thrown errors.
And please, provide a better user-oriented comment to this test case. Example: `"should change the internal state to `done` after data loading in any case."`